lib/model.js

```diff
@@ -2263,7 +2263,8 @@ Model.aggregate = function aggregate() {
 
   aggregate.model(this);
 
-  if (typeof callback === 'undefined') {
+  //TODO find proper way to check that we will need call done()
+  if (typeof callback === 'undefined' && arguments.callee.caller.toString().indexOf(").then") == -1) {
```

Comment:
Really not a big fan of this solution. Would be much better to add a `.then()` function to aggregate.prototype.